### ISSUE TITLE
false positive "unused parameter" warning with argument destructuring

### ISSUE THREAD

**Issue by lovasoa** [intellij-rust/intellij-rust#5524]
## Environment

* **IntelliJ Rust plugin version:** 0.2.123.3132-201
* **Rust toolchain version:** null
* **IDE name and version:** IntelliJ IDEA 2020.1.2 Community Edition (IC-201.7846.76)
* **Operating system:** Linux 4.15.0-101-generic

## Problem description

When a function argument is a destructured struct, it is always marked as "unused".

## Steps to reproduce
```rust
struct Vec2d { x: u32, y: u32 }

fn f(Vec2d { x, y }: Vec2d) -> u32 { x + y }
```
#### Actual
 - **warning** Parameter `x` is never used.
 - **warning** Parameter `y` is never used.

#### Expected

no warning

**Comment by Kobzol** [comment]
Duplicate of https://github.com/intellij-rust/intellij-rust/issues/5109
